infrastructure/src/main/java/org/corfudb/infrastructure/RemoteMonitoringService.java

```diff
@@ -447,11 +449,18 @@ private void correctOutOfPhaseEpochs(PollReport pollReport) {
             // Retrieve the correct layout from quorum of members to reseal servers.
             // If we are unable to reach a consensus from a quorum we get an exception and
             // abort the epoch correction phase.
-            Layout quorumLayout = fetchQuorumLayout(layoutCompletableFutureMap.values()
-                    .toArray(new CompletableFuture[layoutCompletableFutureMap.size()]));
+
+            // TODO(Do we really need a quorum? discover can find a new epoch, without a quorum)
+            // can reuse futures from the method above
+            Layout discoveredLayout = getCorfuRuntime().getLayoutView().discoverLayout(layout.getLayoutServers());
```

Comment:
As you mention in the comment can we reuse the futures, we are going twice to every server for the layout...